### PR TITLE
ERM-1184 Add Edit screen for title instance resources

### PR DESCRIPTION
**Pull request by CalamityC**
* add TitleForm component
* change EResourceEditRoute to differentiate between title and pci
* change EResource component to display the edit button also for a title
* fix translations, displaySuppressFromDiscovery settings, css import
* extend suppress-from-discovery-tests.js for title instance
* add title-instance-edit-test, TitleFormInfo component

### CODE REVIEW

File: src/components/views/EResource.js [hunk 1]
```diff
@@ -79,7 +80,7 @@ const EResource = ({
                       onClick={handlers.onToggleTags}
                     />
                   }
-                  {eresource.class === resourceClasses.PCI &&
+                  { (eresource.class === resourceClasses.PCI || (eresource.class === resourceClasses.TITLEINSTANCE && handlers.isSuppressFromDiscoveryEnabled('title'))) &&
```
Comment (aditya-matukumalli) [inline src/components/views/EResource.js:83]: Re-thinking this, I dont think we should hide the edit button here? As we only have that behavior for a user with no permission to edit. At best, we should probably disable the button with a tooltip explaing the reason or just give them access to the edit pane and the option to close that form (default behavior currently) as none of the fields would be editable in this case.

File: src/components/views/TitleForm.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  static defaultProps = {
+    initialValues: {},
+  }
+
+  constructor(props) {
+    super(props);
+    this.accordionStatusRef = React.createRef();
+  }
+
+  getSectionProps(id) {
```
Comment (aditya-matukumalli) [inline src/components/views/TitleForm.js:49]: Are you using the id thats being passed here?

File: src/components/views/TitleForm.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    const { form, values = {} } = this.props;
+
+    return {
+      form,
+      id,
```
Comment (aditya-matukumalli) [inline src/components/views/TitleForm.js:54]: Same comment as above

File: src/components/views/TitleForm.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+            firstMenu={this.renderFirstMenu()}
+            footer={this.renderPaneFooter()}
+            id="pane-title-form"
+            paneTitle={<FormattedMessage id="ui-agreements.editResource" values={{ name }} />}
+          >
+            <TitleManager record={name}>
+              <form id="form-title">
+                <TitleCardInfo {...this.getSectionProps('info')} title={this.props.eresource} />
+                <div className={css.separator} />
+                <TitleFormInfo isSuppressFromDiscoveryEnabled={isSuppressFromDiscoveryEnabled} name={name} />
```
Comment (aditya-matukumalli) [inline src/components/views/TitleForm.js:155]: You could add all these additional props `title`, `isSuppressFromDiscoveryEnabled` and `name` to the `getSectionProps` function above and just have them passed as `...this.getSectionProps()` to the components.

File: src/components/views/TitleForm.js [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+import { TitleManager } from '@folio/stripes/core';
+import stripesFinalForm from '@folio/stripes/final-form';
+import { checkScope } from '@folio/stripes-erm-components';
+
+import TitleCardInfo from '../TitleCard/TitleCardInfo';
+import TitleFormInfo from './TitleFormInfo';
+
+import css from '../styles.css';
+
+class TitleForm extends React.Component {
```
Comment (aditya-matukumalli) [inline src/components/views/TitleForm.js:23]: Can we change this to be a functional component instead? Its fairly straightforward in this case.

File: src/components/views/TitleFormInfo.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  Col,
+} from '@folio/stripes/components';
+
+const TitleFormInfo = ({
+  isSuppressFromDiscoveryEnabled,
+  name,
+}) => (
+  <div data-test-platform-form-info>
+    <div data-test-title-instance-name>
+      {name}
```
Comment (aditya-matukumalli) [inline src/components/views/TitleFormInfo.js:17]: Do we have to display the name here?

File: src/components/views/TitleFormInfo.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    </div>
+    { isSuppressFromDiscoveryEnabled('title') ?
+      <Col xs={3}>
+        <Field
```
Comment (aditya-matukumalli) [inline src/components/views/TitleFormInfo.js:21]: The checkbox needs to be left aligned

File: src/components/views/TitleFormInfo.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+          name="suppressFromDiscovery"
+          type="checkbox"
+          vertical
+        />
+      </Col> : null
+            }
+  </div>
+);
+
+
```
Comment (aditya-matukumalli) [inline src/components/views/TitleFormInfo.js:34]: whitespace

File: src/routes/EResourceEditRoute.js [hunk 9]
```diff
@@ -7,16 +7,28 @@ import { stripesConnect } from '@folio/stripes/core';
...
-import View from '../components/views/PCIForm';
+import PCIForm from '../components/views/PCIForm';
+import TitleForm from '../components/views/TitleForm';
 import NoPermissions from '../components/NoPermissions';
 import { urls, withSuppressFromDiscovery } from '../components/utilities';
+import { resourceClasses } from '../constants';
 
 class EResourceEditRoute extends React.Component {
   static manifest = Object.freeze({
+    resource: {
```
Comment (aditya-matukumalli) [inline src/routes/EResourceEditRoute.js:18]: Can we change the name to maybe be `eresource` instead. `resource` is a more generic name that is used all across stripes connect and it tends to interfere here.

File: src/routes/EResourceEditRoute.js [hunk 10]
```diff
@@ -38,9 +50,12 @@ class EResourceEditRoute extends React.Component {
       pci: PropTypes.shape({
         PUT: PropTypes.func.isRequired,
       }),
+      title: PropTypes.shape({
+        PUT: PropTypes.func.isRequired,
+      }),
     }).isRequired,
     resources: PropTypes.shape({
-      pci: PropTypes.object,
+      resource: PropTypes.object,
```
Comment (aditya-matukumalli) [inline src/routes/EResourceEditRoute.js:58]: `eresource`

File: src/routes/EResourceEditRoute.js [hunk 11]
```diff
@@ -58,8 +73,8 @@ class EResourceEditRoute extends React.Component {
 
   getInitialValues = () => {
     const { resources } = this.props;
-    const pci = resources?.pci?.records?.[0] ?? {};
-    const initialValues = cloneDeep(pci);
+    const resource = resources?.resource?.records?.[0] ?? {};
```
Comment (aditya-matukumalli) [inline src/routes/EResourceEditRoute.js:76]: `eresource`

File: src/routes/EResourceEditRoute.js [hunk 12]
```diff
@@ -58,8 +73,8 @@ class EResourceEditRoute extends React.Component {
...
+    const initialValues = cloneDeep(resource);
```
Comment (aditya-matukumalli) [inline src/routes/EResourceEditRoute.js:77]: `eresource`

File: src/routes/EResourceEditRoute.js [hunk 13]
```diff
@@ -68,7 +83,7 @@ class EResourceEditRoute extends React.Component {
     this.props.history.push(`${urls.eresourceView(match.params.id)}${location.search}`);
   }
 
-  handleSubmit = (pci) => {
+  handlePCISubmit = (pci) => {
```
Comment (aditya-matukumalli) [inline src/routes/EResourceEditRoute.js:86]: Lets not have 2 different `handleSubmit` functions here. As if we need to extend this functionality to another new `eresource` we would need another function then. Lets have one `handleSubmit` that differentiates between a `pci` and a `title` submit, based on the `eresourceClass`

File: src/routes/EResourceEditRoute.js [hunk 14]
```diff
@@ -83,6 +98,20 @@ class EResourceEditRoute extends React.Component {
       });
   }
 
+  handleTitleSubmit = (title) => {
```
Comment (aditya-matukumalli) [inline src/routes/EResourceEditRoute.js:101]: Same comment as above

File: src/routes/EResourceEditRoute.js [hunk 15]
```diff
@@ -105,7 +139,7 @@ class EResourceEditRoute extends React.Component {
           onClose: this.handleClose,
         }}
         initialValues={this.getInitialValues()}
-        onSubmit={this.handleSubmit}
+        onSubmit={eresourceClass === resourceClasses.TITLEINSTANCE ? this.handleTitleSubmit : this.handlePCISubmit}
```
Comment (aditya-matukumalli) [inline src/routes/EResourceEditRoute.js:142]: With the above changes, this should just be `onSubmit={this.handleSubmit}`

**Review by aditya-matukumalli — CHANGES_REQUESTED** [review]

File: src/components/views/EResource.js (hunk 1 above)
Comment (CalamityC) [reply src/components/views/EResource.js:83]: so what behaviour should I implement? Could we clarify this in the meeting today?

File: src/components/views/TitleFormInfo.js (hunk 6 above)
Comment (CalamityC) [reply src/components/views/TitleFormInfo.js:17]: no, that was probably only a dumb copy-and-paste from another formInfo component

File: src/components/views/TitleFormInfo.js (hunk 7 above)
Comment (CalamityC) [reply src/components/views/TitleFormInfo.js:21]: It is left aligned per default, isn't it?

File: src/routes/EResourceEditRoute.js (hunk 9 above)
Comment (CalamityC) [reply src/routes/EResourceEditRoute.js:18]: sure

File: src/components/views/EResource.js [hunk 16]
```diff
@@ -26,6 +26,7 @@ const propTypes = {
     }),
   }),
   handlers: PropTypes.shape({
+    isSuppressFromDiscoveryEnabled: PropTypes.func.isRequired,
```
Comment (aditya-matukumalli) [inline src/components/views/EResource.js:29]: cleanup proptype as we arent using it anymore

File: src/components/views/TitleForm.js [hunk 17]
```diff
@@ -0,0 +1,154 @@
...
+} from '@folio/stripes/components';
+
+import { AppIcon, TitleManager } from '@folio/stripes/core';
+import stripesFinalForm from '@folio/stripes/final-form';
+import TitleCardInfo from '../TitleCard/TitleCardInfo';
+import TitleFormInfo from './TitleFormInfo';
+
+import css from '../styles.css';
+
+
```
Comment (aditya-matukumalli) [inline src/components/views/TitleForm.js:24]: whitespace

File: src/components/views/TitleForm.js [hunk 18]
```diff
@@ -0,0 +1,153 @@
...
+        </Pane>
+      </Paneset>
+    </HasCommand>
+  );
+};
+
+TitleForm.propTypes = {
+  eresource: PropTypes.object,
+  form: PropTypes.shape({
+    getRegisteredFields: PropTypes.func.isRequired,
```
Comment (aditya-matukumalli) [inline src/components/views/TitleForm.js:133]: Do we need this prop?

**Review by aditya-matukumalli — CHANGES_REQUESTED** [review]

**Review by aditya-matukumalli — APPROVED** [review]

**Pull request merged**